codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/testutil/Rust.kt

```diff
@@ -42,6 +42,37 @@ import java.nio.file.Files.createTempDirectory
 import java.nio.file.Path
 import kotlin.io.path.absolutePathString
 
+// cargo commands and env values
+private object Commands {
+    private const val cfgUnstable = "--cfg aws_sdk_unstable"
+    fun func(s: String, add: String, flag: Boolean): String = if (flag) { "$s $add" } else { s }
+
+    fun cargoEnvDenyWarnings(enableUnstable: Boolean): Map<String, String> {
+        return mapOf(
+            "RUSTFLAGS" to func("-D warnings", cfgUnstable, enableUnstable),
+        )
+    }
+
+    fun cargoEnvAllowDeadCode(enableUnstable: Boolean): Map<String, String> {
+        return mapOf(
+            "RUSTFLAGS" to func("-A dead_code", cfgUnstable, enableUnstable),
+        )
+    }
+
+    private const val allFeature = "--all-features"
+
+    fun cargoTest(enableUnstable: Boolean): String {
+        return func("cargo test", allFeature, enableUnstable)
```

Comment:
Since only features that can be enabled are protected behind feature gate, I thought this would be just as accurate.

I will chang it to `enableAllFeatures`

Reply:
Do you prefer something that allows you to add individual features or a flag that lets you add `--all-festures`?


e.g. 
```kotlin
fun cargoTest(enableAllFeatures: Boolean): String 
```

or 

```kotlin
fun cargoTest(featuresToEnable: Array<String>): String 
```

Reply:
I think we could just have both.

Reply:
Since only features that can be enabled are protected behind feature gate, I thought this is as accurate as it gets.